### DISCUSSION TITLE
Is Serial-Studio available for Raspberry Pi (Raspian)?

### DISCUSSION THREAD

**Discussion by billtubbs** [Serial-Studio/Serial-Studio#81] [Q&A]
I'm not an expert so this may be a dumb question, but can Serial-Studio be installed on a Raspberry Pi?  I see there is a Linux App Image but I don't think that would work on the Raspberry chip would it?  I also have no experience with Qt so no idea if that is an option on Raspberry pi.

I would love to use this if it can be!

**Comment by billtubbs** [comment]
Oh, I see this has already been raised (#37 ). Is there any update on this proposal?

**Comment by alex-spataru** [comment]
Hi, I just managed to compile and install Serial Studio on a Raspberry Pi virtual machine. I have also created a simple install script that automatically:

- Installs build dependencies
- Downloads the source code and compiles it
- Installs the Serial Studio binary on the system

After the script finishes executing, the application can be launched by typing `serial-studio` on the terminal or by navigating to `Applications -> Other -> Serial Studio`. I cannot guarantee performance or functionality, but at least it is a starting point. 😄 

### Script

```bash
#!/bin/bash

#
# Stop script execution if any error occurs
#
set -e

#-----------------------------------------------------------------------
# Install build dependencies
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Updating apt cache..."
echo "/------------------------------------------------------------"
echo ""

# Update package cache
sudo apt update

clear
echo "/------------------------------------------------------------"
echo "/ Install git & C++ build tools..."
echo "/------------------------------------------------------------"
echo ""
sudo apt install git \
                 build-essential

clear
echo "/------------------------------------------------------------"
echo "/ Install Qt5 development packages..."
echo "/------------------------------------------------------------"
echo ""
sudo apt install qt5-qmake \
                 qt5-default \
                 qtbase5-dev \
                 libqt5svg5-dev \
                 qtdeclarative5-dev \
                 libqt5serialport5-dev \
                 qtquickcontrols2-5-dev

clear
echo "/------------------------------------------------------------"
echo "/ Install required QML modules.."
echo "/------------------------------------------------------------"
echo ""
sudo apt install qml-module-qtquick2 \
                 qml-module-qtquick-window2 \
                 qml-module-qtquick-layouts \
                 qml-module-qt-labs-settings \
                 qml-module-qt-labs-platform \
		 qml-module-qtquick-controls \
                 qml-module-qtquick-controls2

#-----------------------------------------------------------------------
# Clone Serial Studio repo
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Downloading Serial Studio source code..."
echo "/------------------------------------------------------------"
echo ""
git clone --recursive https://github.com/serial-studio/serial-studio.git

#-----------------------------------------------------------------------
# Build Serial Studio
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Building Serial Studio..."
echo "/------------------------------------------------------------"
echo ""

cd serial-studio
mkdir build
cd build
qmake ../ CONFIG+=release DEFINES+=SERIAL_STUDIO_SOFTWARE_RENDERING
make

#-----------------------------------------------------------------------
# Install Serial Studio
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Installing Serial Studio..."
echo "/------------------------------------------------------------"
echo ""

sudo make install

#-----------------------------------------------------------------------
# Delete source code
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Removing source code directory..."
echo "/------------------------------------------------------------"
echo ""

cd ..
cd ..
rm -rf serial-studio

#-----------------------------------------------------------------------
# Run Serial Studio
#-----------------------------------------------------------------------

clear
echo "/------------------------------------------------------------"
echo "/ Executing Serial Studio..."
echo "/------------------------------------------------------------"
echo ""

serial-studio
```

### Steps to run the script

1. Copy and paste the above script in a text editor.
2. Save the generated file as `serial-studio-install.sh`
3. Open a terminal, navigate to the directory in which you saved the script and run:

```bash
chmod +x serial-studio-install.sh
./serial-studio-install.sh
```

Here is a screenshot of the terminal commands required to execute the script:

<img width="654" alt="exec-script" src="https://user-images.githubusercontent.com/4225542/147885390-56da987b-8b83-4ad8-baf5-a891b7fce328.png">

After running the script, you might be promoted to introduce your password to install the build dependencies. Also, make sure to type `y` when the `apt install` prompts you if you want to continue installing the packages:

<img width="654" alt="exec-script" src="https://user-images.githubusercontent.com/4225542/147885467-6c9b44e2-073e-46fa-a913-6cae828e0ebc.png">

After installing dependencies, the script shall download the latest commit from GitHub & begin compiling Serial Studio. This might take a while depending on your hardware setup. If everything goes ok, you should be able to see a Serial-Studio window at the end of the installation process:

<img width="1536" alt="serial-studio-raspberry" src="https://user-images.githubusercontent.com/4225542/147885512-fbe5b920-d09c-40c7-b94b-303c93d06a93.png">

Please let me know if this helps you, happy new year btw!

**Reply by billtubbs** [reply]
Awesome, thanks!  I will try this out as soon as my RPi is back from the field.

**Reply by alex-spataru** [reply]
Thanks a lot! Please let me know if it works!

**Comment by g7kse** [comment]
I thought I'd give this a go. There is a slight problem with qt5-default. It doesn't seem to be in the current repo so will throw up an error. After a bit of googling about using the code below solves it
```
sudo apt install qtbase5-dev qtchooser qt5-qmake qtbase5-dev-tools
```
I just removed the reference to the qt5-default from the script (for now) and it built fine. Its running nicely on a RPi 3 with and old slow SD card and is receiving data fine. 

Nice job Alex

**Comment by Redika-ak** [comment]
everything was running fine until this code appeared while compiling was in progress
FATAL: qstringconverter: No such file or directory

what's the solution?

**Comment by Maxlma** [comment]
Thanks for the Script i fixed it replacing the qt-5 lines and it start to install, as suggested but unfortunately now is giving me a Bluetooth error, any idea how to fix or at least skip it ?
" Info: creating stash file /home/nexus/serial-studio/build/.qmake.stash
Project ERROR: Unknown module(s) in QT: svg quick bluetooth serialport quickcontrols2
 "

**Reply by jlgarciamx** [reply]
I have exactly the same error, can somebody provide feedback? Thank you in advance

**Reply by jlgarciamx** [reply]
Alex, I'm trying to run Serial Studio in a Raspberry-pi 4B, I'm following your instructions but after the following instruction:
qmake ../ CONFIG+=release DEFINES+=SERIAL_STUDIO_SOFTWARE_RENDERING

I'm getting the following error :
Project ERROR: Unknown module(s) in QT: bluetooth

Do you have any clue how to solve it? Do I need to install something else in the Raspberry-Pi (like QT)? Thank you very much in advance.
